harness/determined/exec/gc_checkpoints.py

```diff
@@ -49,15 +54,15 @@ def main(argv: List[str]) -> None:
     )
     parser.add_argument(
         "--experiment-config",
-        type=json.loads,
+        type=json_file_arg,
         default=os.getenv("DET_EXPERIMENT_CONFIG", {}),
```

Comment:
thought: The logic of how `argparse` handles defaults is a bit weird -- it returns the given `default` unchanged [if is a string](https://github.com/python/cpython/blob/master/Lib/argparse.py#L2082) or passes it through the `type` function otherwise. To reduce [WTFs/m](https://www.osnews.com/story/19266/wtfsm/) for anyone looking at this in the future, perhaps just change the defaults to the currently used paths, which would also simplify the invocation in the master. On the other hand, that may be slightly less useful as a behavior on this end, since now there must exist a populated file. On the [gripping hand](https://scifi.stackexchange.com/a/398), we never use that default behavior in practice anyway.

Reply:
I'd really, really like to keep defaults out of the obscure, dusty, places of the code, which `gc_checkpoints.py` definitely counts as.

I could just make these required positional arguments?

The reason I didn't already address this was that the default behavior was obviously not supported anyway; there's no support for an empty experiment config whatsoever.  I assumed the `default` behavior was not meant to provide a functional default but just to give cli args priority over the environment variables.

IMO my change did not make that situation any better or worse.  But I'm happy to remove crusty old features that are totally unused.  There's several parts of gc_checkpoints.py which qualify.

Reply:
Yep, I'd be totally fine with either leaving it as it is or making them positional. Though, if you have other stuff in mind, I'd weakly lean toward doing all the refactoring together in a separate PR.

> the obscure, dusty, places of the code, which gc_checkpoints.py definitely counts as.

Shh, you'll hurt its feelings! It's right here!

Reply:
Ok, in that case, I'll leave it as is.  While I do have things I'd like to remove, I'd lean towards not fixing what is clearly not broken, especially when this cleanup would make literally nothing else in the system any easier or harder.